server/service/src/processors/transfer/requisition/test.rs

```diff
@@ -408,34 +429,48 @@ impl RequisitionTransferTester {
 
         let response_req = self.response_requisition.clone().unwrap();
 
-        let approved_response_requisition = RequisitionRow {
-            approval_status: Some(ApprovalStatusType::Approved),
-            ..response_req
-        };
-
-        requisition_repo
-            .upsert_one(&approved_response_requisition)
-            .unwrap();
-
         let response_lines = RequisitionLineRepository::new(&ctx.connection)
             .query_by_filter(
                 RequisitionLineFilter::new()
-                    .requisition_id(EqualFilter::equal_to(&approved_response_requisition.id)),
+                    .requisition_id(EqualFilter::equal_to(&response_req.id)),
             )
             .unwrap();
 
+        let line_count = response_lines.len();
+        let mut i = 1;
         for line in response_lines {
             let mut line = line.requisition_line_row;
+            log::debug!(
+                "{}: Updating line {} to approved (42.0) {}/{}",
+                thread_number,
+                line.id,
+                i,
+                line_count
+            );
             line.approved_quantity = 42.0;
             RequisitionLineRowRepository::new(&ctx.connection)
                 .upsert_one(&line)
                 .unwrap();
+            i += 1;
         }
 
+        let approved_response_requisition = RequisitionRow {
```

Comment:
This is the actual fix, setting the status after we're already updated the lines...